sqlmesh/core/scheduler.py

```diff
@@ -934,13 +987,27 @@ def _check_ready_intervals(
         if not (signals and signals.signals_to_kwargs):
             return intervals
 
+        signal_names = signals.signals_to_kwargs.keys()
+
+        if (
```

Comment:
This check is quite brittle. What if there are multiple signals one of which is `freshness`?